packages/SwingSet/src/controller/startXSnap.js

```diff
@@ -62,32 +88,62 @@ export function makeStartXSnap(options) {
     netstringMaxChunkSize: NETSTRING_MAX_CHUNK_SIZE,
   };
 
+  /** @param {StartXSnapInitDetails} initDetails */
+  function getSnapshotLoadOptions(initDetails) {
+    switch (initDetails.from) {
+      case 'bundles':
+        return undefined;
+
+      case 'snapStore': {
+        if (!snapStore) {
+          // Fallback to load from bundles
+          return undefined;
```

Comment:
note for future: this should be an error, if the caller expected a `load-snapshot` but the implementation somehow failed to provide a `snapStore` (which probably isn't even a thing anymore, the notion that `snapStore` was optional is pretty old), then a blank worker is not a functional substitute

Reply:
of course, this whole clause will go away if/when the vat-loader gives us `from: 'snapshotStream'` instead of `from: 'snapStore'`

Reply:
Right, but the goal of this commit was as a pure refactor, and some tests seemed to rely on this behavior.

Hopefully we can clean all that up in a follow up.